exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -209,6 +187,44 @@ func (ampmr *AzureMachinePoolMachineController) Reconcile(ctx context.Context, r
 		return reconcile.Result{}, err
 	}
 
+	// Note: this is critical for deletion!
```

Comment:
can you explain why?

Reply:
This is because we need to check to see if the resource group is being deleted, and if so, we just go ahead and remove the finalizer since deletion can't finish until the finalizer is gone. I moved this up here as when we are deleting the RG, the AzureMachinePool is gone by the time we reach this code, and the AMPM reconciler gets stuck requeuing as it can't construct the scope when the AzureMachinePool is gone.

Reply:
> When the RG is being deleted, the AzureMachinePool gets deleted before the AzureMachinePoolMachines as the AMP removes the finalizer, bypassing the usual behavior of waiting for the children to be deleted

Are the AMPMs not owned by the AzureMachinePool or the MachinePool? if so, k8s garbage collection should take care of blocking the AMP deletion until all dependent AMPMs are deleted, regardless of CAPZ deleting its own controller finalizer...

https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#owner-references-in-object-specifications